dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/responses/McpResponseAdapter.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.responses;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import jakarta.json.JsonValue;
+import jakarta.json.bind.adapter.JsonbAdapter;
+
+/**
+ *
+ */
+public class McpResponseAdapter implements JsonbAdapter<McpResponse, JsonObject> {
```

Comment:
May not need this:
- we aren't doing conversions from JSON to responses
- we shouldn't need it for converting to JSON - jsonb should handle everything for us based on the properties of the object